Added supported device: Edimax smart plugs

rkabadi opened the pull request:
Added support for the folllowing devices
- SP-1101W
- SP-2101W

Device working
![image](https://cloud.githubusercontent.com/assets/5388441/8892201/935b2404-3301-11e5-8121-50ac53aabc28.png)

Added submodule: https://github.com/rkabadi/pyedimax (README still needed)

Look forward to feedback!

rmkraus committed:
Light update to PyISY version 1.0.5

This is a lighter update to version 1.0.5 to fix Issue #201 more
immediately.

balloob committed:
Merge pull request #208 from rmkraus/isy_dev_master

Light update to PyISY version 1.0.5

balloob committed:
Merge pull request #202 from balloob/dev

Update master with latest changes

balloob committed:
Merge branch 'dev'

rkabadi committed:
- Added git submodule @ https://github.com/rkabadi/pyedimax
- Added edimax.py module to interface with Edimax SP-1101W and SP-2101W

rkabadi committed:
- Added for smartplug
- Added error check for host param in config.yaml
- Fixed SmartPlugSwitch is_on method
- Edimax smartplug works now!

rkabadi committed:
- Added support for getting power on SP2101W devices (returns None on SP1101W)

balloob commented:
Awesome! I see you used a git submodule, we're in the progress of moving
away from it. Add a setup.py to your git repo and add the github url as
requirements.

See net disco as example.

Also, please exclude the updates to the non related git submodule from your
PR.

On Sun, Jul 26, 2015, 04:18 rkabadi <notifications@github.com> wrote:

> Added support for the folllowing devices
>
>    - SP-1101W
>    - SP-2101W
>
> Device working
> [image: image]
> <https://cloud.githubusercontent.com/assets/5388441/8892201/935b2404-3301-11e5-8121-50ac53aabc28.png>
>
> Added submodule: https://github.com/rkabadi/pyedimax (README still needed)
>
> Look forward to feedback!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/balloob/home-assistant/pull/229
> Commit Summary
>
>    - Light update to PyISY version 1.0.5
>    - Merge pull request #208 from rmkraus/isy_dev_master
>    - Merge pull request #202 from balloob/dev
>    - Merge branch 'dev'
>    - - Added git submodule @ https://github.com/rkabadi/pyedimax
>    - - Added for smartplug
>    - - Added support for getting power on SP2101W devices (returns None
>    on SP1101W)
>
> File Changes
>
>    - *M* .coveragerc
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-0> (1)
>    - *M* .gitmodules
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-1> (3)
>    - *M*
>    homeassistant/components/frontend/www_static/polymer/home-assistant-js
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-2> (2)
>    - *A* homeassistant/components/switch/edimax.py
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-3> (77)
>    - *M* homeassistant/external/netdisco
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-4> (2)
>    - *A* homeassistant/external/pyedimax
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-5> (1)
>    - *M* homeassistant/external/pymysensors
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-6> (2)
>    - *M* homeassistant/external/pynetgear
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-7> (2)
>    - *M* homeassistant/external/pywemo
>    <https://github.com/balloob/home-assistant/pull/229/files#diff-8> (2)
>
> Patch Links:
>
>    - https://github.com/balloob/home-assistant/pull/229.patch
>    - https://github.com/balloob/home-assistant/pull/229.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/229>.
>

balloob commented:
All the code looks fine, no comments there!

rkabadi committed:
- Reverted submodule updates

rkabadi committed:
- Reverted submodule updates. This is the 2nd attempt since the first one did not work

rkabadi committed:
- Removed https://github.com/rkabadi/pyedimax as submodule
- Added https://github.com/rkabadi/pyedimax to requirements
- Modified edimax.py to import pyedimax from python3 default packages

rkabadi commented:
@balloob, I appreciate the feedback! Took your recommendations into account as follows:
- Reverted submodule folder updates
- Added setup.py to https://github.com/rkabadi/pyedimax. Moved libraries to pyedimax package
- Added https://github.com/rkabadi/pyedimax.git in requirements.txt

Retested with deleted submodule and all works as expected.

rmkraus commented on homeassistant/components/switch/edimax.py:
```diff
@@ -0,0 +1,75 @@
+
+    if not host:
+        logging.getLogger(__name__).error(
+            'Missing config variable %s', CONF_HOST)
```
The preferred way of doing this is by using homeassistant.helpers.validate_config

rmkraus commented on homeassistant/components/switch/edimax.py:
```diff
@@ -0,0 +1,75 @@
+    def name(self):
+        """ Returns the name of the Smart Plug, if any. """
+        #TODO: dynamically get name from device using requests
+        return 'Edimax Smart Plug'
```
Todo comments are not accepted into the repository. Please change this to future. It may be nice to at least have a config option to set the name of the device if the name is not pulled from the device.

rmkraus commented:
I added a few more comments. Also, it looks like you'll need to merge with the latest dev branch.

rkabadi committed:
Used validate_config to ensure 'host' parameter in edimax config. Added name option to edimax config

rkabadi committed:
Merge remote-tracking branch 'upstream/dev' into edimax_smart_plug

Conflicts:
	requirements.txt

rkabadi committed:
Updated branch to avoid conflicts in requirements.txt

rkabadi commented:
@rmkraus, I made some changes from your suggestions. Need to fix a flake8 violation from my latest push

rkabadi committed:
Fixed flake8 violations

rmkraus commented on homeassistant/components/switch/edimax.py:
```diff
@@ -0,0 +1,86 @@
+
+# setup logger
+_LOGGER = logging.getLogger(__name__)
+
```
Please add a REQUIREMENTS list here. Something like:
```
REQUIREMENTS = ['https://github.com/rkabadi/pyedimax/archive/master.zip']
```

rmkraus commented:
Almost there! I added one more comment. After that, I can't think of anything more to nitpick :wink: .

rkabadi committed:
Added REQUIREMENTS lilst to edimax.py

rmkraus commented:
I think you may need to merge with upstream/dev again.

rmkraus commented:
I think you may need to merge with upstream/dev again.

balloob commented:
Awesome, merged!

fabaff commented:
This is the documentation:
https://github.com/balloob/home-assistant.io/commit/b8d85702834ceb3f84c82bbc53fe7d4063840821
